### DISCUSSION TITLE
Next 12: Difference between .client.js components and "universal" components?

### DISCUSSION THREAD

**Discussion by dalmo3** [vercel/next.js#30497] [App Router]
After reading the blog announcement and documentation on RSC, I couldn't find any reason to use `.client.js` components other than readability.

The docs mention that _a server component can not be imported by a client component_, but when playing around with it I was able to do just that: https://codesandbox.io/s/flamboyant-oskar-piyqb?file=/pages/2sc.client.js

In fact, the opposite of that - _a client component can not be imported by a server component_ - is more likely to be the intended outcome, since you can't use hooks in server components.

**Comment by IshanKBG** [comment]
I guess that .client components runs on client side whereas universal components can be used in both server and client components for reusablity

**Reply by dalmo3** [reply]
That's the theory, but in the sandbox I linked I was able to import a .client component inside a server component (see `2cs.server.js`)
